### PR TITLE
GHA setup-linux should always be pair with teardown-linux

### PR DESCRIPTION
**Pull request by huydhn**
This explains the existence of a running container `pytorch-linux-focal-py3.8-gcc7`.  For example,  https://github.com/pytorch/pytorch/actions/runs/5138349666/jobs/9344366066 on `i-097fac5d9b9bef249`.  The immediate job running before on this runner was a doc build jobs:

```
{
    "name": "linux-docs / build-docs-functorch-false",
    "html_url": "https://github.com/pytorch/pytorch/actions/runs/5184821352/jobs/9344262329",
    "_event_time": "2023-06-06T05:21:23.569553Z",
    "runner_name": "i-097fac5d9b9bef249",
    "head_branch": "gh/titaiwangms/25/head",
    "head_sha": "5d48fd183875e4eea44a94df239eae356e852939",
    "workflow_name": "pull",
    "conclusion": "success",
    "line": null
}
```

This might be related to OOM issue happening on these runners.

Thoughts:

* May be we should have a linter for this to make sure that GHA setup-OS always pairs with teardown-OS
* Nova GHA `setup-linux` is updated by https://github.com/pytorch/test-infra/pull/4275

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: .github/workflows/_calculate-docker-image.yml [hunk 1]
```diff
@@ -39,3 +39,7 @@ jobs:
           # longer term fix would be to refactor linter so that building the new
           # image and running lint are done in the same runner like _linux-build
           skip_push: false
+
+      - name: Teardown Linux
+        uses: pytorch/test-infra/.github/actions/teardown-linux@main
+        if: always()
```
Comment (malfet) [inline .github/workflows/_calculate-docker-image.yml:45]: I'm not sure we want that, as it is also called as part of the build, isn't it?

Comment (malfet) [reply .github/workflows/_calculate-docker-image.yml:45]: Actually, it's current called only from lint.yml (rest are calling action), so should we just fold it in there?

Comment (huydhn) [reply .github/workflows/_calculate-docker-image.yml:45]: Ah no, that is https://github.com/pytorch/pytorch/blob/main/.github/actions/calculate-docker-image/action.yml GHA while this one is https://github.com/pytorch/pytorch/blob/main/.github/workflows/_calculate-docker-image.yml workflow.  The workflow one is a wrapper used by lint https://github.com/pytorch/pytorch/blob/17737f9d0e683a01b1873e9b7f948de9035e4951/.github/workflows/lint.yml#L17-L21 as a reusable workflow

Comment (huydhn) [reply .github/workflows/_calculate-docker-image.yml:45]: > Actually, it's current called only from lint.yml (rest are calling action), so should we just fold it in there?

Sound good.  Let me see if I can move it into lint and clean up the workflow

**Review by ZainRizvi — APPROVED** [review]

**Review by atalman — APPROVED** [review]

**Pull request closed without merging**